sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/StringLiteralShouldNotBeDuplicated.cs

```diff
@@ -51,55 +51,45 @@ public sealed class StringLiteralShouldNotBeDuplicated : ParameterLoadingDiagnos
 
         protected override void Initialize(ParameterLoadingAnalysisContext context)
         {
-            context.RegisterCompilationStartAction(
-                csac =>
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                c =>
                 {
-                    if (csac.Compilation.IsTest())
```

Comment:
The RSPEC says this is a "main sources" rule. Is this check for test code no longer required because it is being done in the base class?

Reply:
Yes this is done on base class.